pages/[id]/bridge.tsx

```diff
@@ -299,12 +306,25 @@ const Bridge = () => {
               </div>
               <div className="text-center">
                 {mounted &&
-                /* ⚪️ First, if we are on the wrong network, prompt to switch networks. */
+                /* ⚪️ First, if we are on the wrong network, prompt to switch networks. Add new network to user wallet if needed. */
                 source.chain.id !== chain?.id ? (
                   <button
                     type="button"
                     className="mt-5 rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-gray-300 disabled:cursor-not-allowed"
-                    onClick={() => walletClient?.switchChain({ id: source.chain.id })}
+                    onClick={async () => {
```

Comment:
We also switch chains on the delegate page. It would be good to abstract this to a utility function and reuse on that page

Reply:
Made a custom hook in [d459419](https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/133/commits/d4594196ee44dbf420cbe2d9a0d141a5655889b3)

Reply:
Actually added the function to `@util` [2820cbc](https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/133/commits/2820cbca985755420de837fe2d3b2c98656cc2ec) because `React Hooks must be called in a React function component or a custom React Hook function` error. Lmk if I did it right.